File: pueblo_golf_tournament_web/lib/models/tournament.dart
import 'package:json_annotation/json_annotation.dart';
import 'base-model.dart';
part 'tournament.g.dart';

@JsonSerializable()
class Tournament extends BaseModel {
  final String name;
  final String? bannerPhoto;
  final DateTime startDate;
  final DateTime endDate;
  final String? venuePlace;
  final String? venueAddress;
  final String? venueCity;
  final String? venueCountry;
  final int numberOfSlots;

  factory Tournament.fromJson(Map<String, dynamic> json) =>
      _$TournamentFromJson(json);

  Tournament(
      {required this.name,
      required this.bannerPhoto,
      required this.startDate,
      required this.endDate,
      this.venuePlace,
      this.venueAddress,
      this.venueCity,
      this.venueCountry,
      required this.numberOfSlots,
      required super.id});
  Map<String, dynamic> toJson() => _$TournamentToJson(this);
}

File: pueblo-golf-tournament-api/Entities/Tournament.cs
using pueblo_golf_tournament_api.Base;

namespace pueblo_golf_tournament_api.Entities
{
    public class Tournament : BaseEntity
    {
        public string? Name { get; set; }
        public string? BannerPhoto { get; set; }
        public DateTime StartDate { get; set; }
        public DateTime EndDate { get; set; }
        public double RegistrationFee { get; set; }
        public string? VenuePlace { get; set; }
        public string? VenueAddress { get; set; }
        public string? VenueCity { get; set; }
        public string? VenueCountry { get; set; }
        public int NumberOfSlots { get; set; }
        public string? NameOfTournamentSupport { get; set; }
        public string? ContactNumberOfTournamentSupport { get; set; }

    }
}